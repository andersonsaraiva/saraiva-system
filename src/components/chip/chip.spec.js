import { shallowMount } from '@vue/test-utils';
import Comp from './chip.vue';

describe('chip.vue', () => {
  it('matches snapshot', () => {
    const wrapper = shallowMount(Comp);
    expect(wrapper.element).toMatchSnapshot();
  });

  it('verifica se Ã© emitido ao clicar no icone', async () => {
    const wrapper = shallowMount(Comp);

    await wrapper.setProps({ removable: true });
    await wrapper.vm.remove();

    expect(wrapper.emitted().remove[0]).toEqual([]);
    expect(wrapper.element).toMatchSnapshot();
  });
});
