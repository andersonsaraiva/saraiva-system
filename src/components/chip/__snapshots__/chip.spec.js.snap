// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`chip.vue matches snapshot 1`] = `
<span
  class="on-chip on-chip-size-base on-chip-medium on-chip-round"
>
  
  
  <!--v-if-->
</span>
`;

exports[`chip.vue verifica se Ã© emitido ao clicar no icone 1`] = `
<span
  class="on-chip on-chip-size-base on-chip-medium on-chip-removable on-chip-round"
>
  
  
  <button
    class="on-chip-close"
    tabindex="-1"
  >
    X
  </button>
</span>
`;
